pymeasure/instruments/tcpowerconversion/tccxn.py

```diff
@@ -0,0 +1,392 @@
+        if header[0] != 82:
+            raise ValueError(f"invalid header start byte '{header[0]}' received")
+        if header[1] != self.address:
+            raise ValueError(f"invalid address byte '{header[1]}' received; "
+                             f"should be {self.address.to_bytes(1, 'big')}")
+        datalength = int.from_bytes(header[2:], "big")
+        data = super().read_bytes(datalength)
+        chksum = super().read_bytes(2)
+        if chksum == self._checksum(header + data):
+            return data
+        else:
+            raise ValueError(
+                f"checksum error in received message {header + data} "
+                f"with checksum {self._checksum(header + data)} "
+                f"but received {chksum}")
+
+    def write(self, command):
+        """
+        Writes to the instrument and includes needed command header/address
+
+        :param command: command string to be sent to the instrument
+        """
+        fullcmd = self._prepend_cmdheader(command.encode())
+        super().write_bytes(fullcmd + self._checksum(fullcmd))
+        self.check_acknowledgment()
+
+    def check_acknowledgment(self):
+        """
+        check reply string for acknowledgement string
+        """
+        ret = super().read_bytes(1)
+        if ret == b"*":
+            return
+        # no valid acknowledgement message found
+        raise ValueError(
+            f"invalid reply '{ret}' found in acknowledgement check")
+
+    id = Instrument.measurement(
+        "Gi\x00\x01\x00\x00",
+        """ Device identification string """,
+        cast=str,
+        get_process=lambda d: d.decode()[2:-1].strip(),
+    )
+
+    serial = Instrument.measurement(
+        "Gi\x00\x02\x00\x00",
+        """ Serial number of the instrument """,
+        cast=str,
+        get_process=lambda d: d.decode()[2:-1].strip(),
+    )
+
+    firmware_version = Instrument.measurement(
+        "Gf\x00\x00\x00\x00",
+        """ UI-processor and RF-processor firmware version numbers """,
+        preprocess_reply=lambda d: struct.unpack("BBBB", d),
+        get_process=lambda v: str.format("UI {}.{}, RF {}.{}", *v)
+    )
+
+    pulse_params = Instrument.measurement(
+        "GE\x00\x00\x00\x00",
+        """ Get pulse on/off time of the pulse waveform """,
+        preprocess_reply=lambda d: struct.unpack(">HH", d),
+    )
+
+    frequency = Instrument.measurement(
+        "GF\x00\x00\x00\x00",
+        """ Get operating frequency in Hz""",
+        preprocess_reply=lambda d: struct.unpack(">L", d),
+    )
+
+    power = Instrument.measurement(
+        "GP\x00\x00\x00\x00",
+        """ Get power readings for forward/reverse/load power in watts """,
+        preprocess_reply=lambda d: struct.unpack(">HHH", d),
+        get_process=lambda d: (float(d[0])/10, float(d[1])/10, float(d[2])/10),
+    )
+
+    status = Instrument.measurement(
+        "GS\x00\x00\x00\x00",
+        """ Get status field, where used bits correspond to:
```

Comment:
If you want, you could use an `IntFlag` enum to implement these status fields, to get a more user-readable representation back. Check other uses of `IntFlag` in the codebase to see how that's done.

Reply:
There are two issues regarding possible implementations: #733 #736

Reply:
Thanks for this hint. This looks very useful to me. I tried to implement it.